runtime/stdlib/test-framework.go

```diff
@@ -73,6 +73,8 @@ type TestFramework interface {
 	Reset(uint64)
 
 	MoveTime(int64)
+
+	NewEmulatorBackend() TestFramework
```

Comment:
`TestFramework` returning another `TestFramework` seems a bit strange/can be confusing.

Instead, could we split the current `TestFramework` into two:

```go
type TestFramework interface {

    NewEmulatorBackend() Blockchain

    ReadFile(string) (string, error)
}

type Blockchain interface {

    RunScript(...) *ScriptResult

    CreateAccount() (*Account, error)

    AddTransaction(...) error

    ExecuteNextTransaction() *TransactionResult

    CommitBlock() error

    DeployContract( ... )

    ...
}
```

Reply:
Yeah, good point, this was feeling little bit counter-intuitive in the tests as well :sweat_smile: 
I will update it :+1:

Reply:
I have been thinking about moving all these into the `cadence-tools/test` repository, if it's possible. I haven't checked if all the methods involved are actually publicly exposed.
What are your thoughts on this? I like the fact that Cadence exposes interfaces to be implemented by various providers, but in this case, the reality is that there's only one provider, and this somewhat slows down development. Certainly out of scope of this PR, just thought of bringing it up.

Reply:
The problem is that the standard libraries and the `Test` contract implementation is in the `cadence` repo. So moving this interface to `cadence-tools` would create a circular dependency between the two repos. That was the main reason to have this interface inside this repo.

Reply:
I have extracted the `Blockchain` interface out of the `TestFramework` interface here: https://github.com/onflow/cadence/pull/2744/commits/6862627915b88a3c64d8de99dc34ffdd0c788616